lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1159,11 +1133,21 @@ static void optimizeArithmeticNodes(Function *F) {
     Node *node = worklist.back();
     worklist.pop_back();
 
-    auto *sn = simplifyNode(node, F);
-    if (sn != node) {
-      node->getNthResult(0).replaceAllUsesOfWith(sn);
+    auto *SN = simplifyNode(node, F);
+    if (SN != node) {
+      node->getNthResult(0).replaceAllUsesOfWith(SN);
       // The simplified node could be further simplified.
-      worklist.push_back(sn);
+      worklist.push_back(SN);
+
+      // The simplified node's operands could be further simplified as well.
+      if (SN->isArithmetic()) {
+        auto *LHS = SN->getNthInput(0).getNode();
+        auto *RHS = SN->getNthInput(1).getNode();
```

Comment:
Could we iterate over all the inputs instead of hardcoding the number of operands?